Polyclinic/Polyclinic.Application/Service/DoctorService.cs

```diff
@@ -0,0 +1,121 @@
+    IRepository<Doctor, int> doctorRepository,
+    IRepository<Appointment, int> appointmentRepository,
+    IRepository<Specialization, int> specializationRepository,
+    IMapper mapper) : IDoctorService
+{
+    /// <summary>
+    /// Creates a new doctor with specified specialization
+    /// </summary>
+    /// <param name="dto">Data for creating the doctor including specialization ID</param>
+    /// <returns>Created doctor with specialization details</returns>
+    /// <exception cref="ArgumentException">Thrown when specialization with specified ID is not found</exception>
+    public DoctorDto Create(DoctorCreateUpdateDto dto)
+    {
+        var specialization = specializationRepository.Read(dto.SpecializationId);
+        if (specialization == null)
+            throw new ArgumentException($"Specialization with ID {dto.SpecializationId} not found");
+
+        var newDoctor = mapper.Map<Doctor>(dto);
+        newDoctor.Specialization = specialization; 
+
+        var lastDoctor = doctorRepository.ReadAll().OrderByDescending(d => d.Id).FirstOrDefault();
+        newDoctor.Id = (lastDoctor?.Id ?? 0) + 1;
+
+        doctorRepository.Create(newDoctor);
+        return mapper.Map<DoctorDto>(newDoctor);
+    }
+
+    /// <summary>
+    /// Deletes a doctor by identifier
+    /// </summary>
+    /// <param name="dtoId">Doctor identifier</param>
+    /// <returns>True if deletion was successful</returns>
+    public bool Delete(int dtoId)
+    {
+        doctorRepository.Delete(dtoId);
+        return true;
+    }
+
+    /// <summary>
+    /// Retrieves a doctor by identifier
+    /// </summary>
+    /// <param name="dtoId">Doctor identifier</param>
+    /// <returns>Doctor if found</returns>
+    public DoctorDto? Get(int dtoId) =>
+        mapper.Map<DoctorDto?>(doctorRepository.Read(dtoId));
+
+    /// <summary>
+    /// Retrieves all doctors
+    /// </summary>
+    /// <returns>List of all doctors</returns>
+    public List<DoctorDto> GetAll() =>
+        mapper.Map<List<DoctorDto>>(doctorRepository.ReadAll());
+
+    /// <summary>
+    /// Updates an existing doctor with new data including specialization
+    /// </summary>
+    /// <param name="dto">Data for updating the doctor including specialization ID</param>
+    /// <param name="dtoId">Doctor identifier</param>
+    /// <returns>Updated doctor with specialization details</returns>
+    /// <exception cref="ArgumentException">Thrown when doctor or specialization with specified ID is not found</exception>
+    public DoctorDto Update(DoctorCreateUpdateDto dto, int dtoId)
+    {
+        var specialization = specializationRepository.Read(dto.SpecializationId);
+        if (specialization == null)
+            throw new ArgumentException($"Specialization with ID {dto.SpecializationId} not found");
+
+        var existingDoctor = doctorRepository.Read(dtoId);
+        if (existingDoctor == null)
+            throw new ArgumentException($"Doctor with ID {dtoId} not found");
```

Comment:
Аналогично можно упростить:
```suggestion
        var specialization = specializationRepository.Read(dto.SpecializationId)
          ?? throw new ArgumentException($"Specialization with ID {dto.SpecializationId} not found");
      var existingDoctor = doctorRepository.Read(dtoId)
          ?? throw new ArgumentException($"Doctor with ID {dtoId} not found");
```